False negative in SQL injection detection (S608) when selecting multiple columns across multiple lines

Here's a file that demonstrates the issue when checked with `ruff check --select S608 test.py`:

```py
user_input = "SELECT * FROM bad_stuff"
bad_detected = f"""
    SELECT *, foo
    FROM ({user_input}) raw
"""

bad_undetected = f"""
    SELECT *,
        foo
    FROM ({user_input}) raw
"""
```

I'd expect both `bad_detected` and `bad_undetected` to be warned about, but this only warns with `test.py:2:16: S608 Possible SQL injection vector through string-based query construction`.

This used to work correctly in ruff 0.4, but broke at some point between then and ruff 0.9.2.

--- a/crates/ruff_linter/resources/test/fixtures/flake8_bandit/S608.py
+++ b/crates/ruff_linter/resources/test/fixtures/flake8_bandit/S608.py
@@ -154,3 +154,15 @@ FROM {var}.table
 # to be handled separately
 # query58 = f"SELECT\
 #  * FROM {var}.table"
+
+
+# https://github.com/astral-sh/ruff/issues/15653
+query59 = f"""
+    SELECT *, foo
+    FROM ({user_input}) raw
+"""
+query60 = f"""
+    SELECT *,
+        foo
+    FROM ({user_input}) raw
+"""

--- a/crates/ruff_linter/src/rules/flake8_bandit/rules/hardcoded_sql_expression.rs
+++ b/crates/ruff_linter/src/rules/flake8_bandit/rules/hardcoded_sql_expression.rs
@@ -12,7 +12,16 @@ use crate::checkers::ast::Checker;
 use crate::Locator;
 
 static SQL_REGEX: LazyLock<Regex> = LazyLock::new(|| {
-    Regex::new(r"(?i)\b(select\s+.*\s+from\s|delete\s+from\s|(insert|replace)\s+.*\s+values\s|update\s+.*\s+set\s)")
+    Regex::new(
+        r"(?isx)
+        \b
+        (select\s+.*\s+from\s
+        |delete\s+from\s
+        |(insert|replace)\s+.*\s+values\s
+        |update\s+.*\s+set\s
+        )
+    ",
+    )
     .unwrap()
 });
 

